### PR TITLE
Add a pull request description template

### PR DESCRIPTION
**Pull request by xylar**
This merge adds a template for new Omega PRs.

### CODE REVIEW

**Comment by xylar** [comment]
@philipwjones, @sarats, @mark-petersen, @vanroekel and others, I mostly want this PR to open up a conversation about what we might want as a checklist for each Omega PR.  What I have here is based on my experience with compass and polaris.  I'm sure you'll have some changes you'll want to make.

File: .github/pull_request_template.md
Comment (xylar) [inline .github/pull_request_template.md:13]: This obviously doesn't exist yet, so I would be fine with removing it for now.

**Comment by sarats** [comment]
Overall looks good to me.

Description should include tests added/modified/impacted.
Perhaps encouraging folks to add a CDash build result when ticking the checklist would be nice. 

Only for performance related PRs, I want to see a PACE link before/after.

**Comment by xylar** [comment]
@sarats, go ahead and edit to add these where you think it appropriate in the check list.

**Review by sarats — APPROVED** [review]

**Review by philipwjones — APPROVED** [review]
This looks good to me now.

**Comment by xylar** [comment]
@mark-petersen and @vanroekel, still waiting for your review.

**Review by vanroekel — APPROVED** [review]
This check list looks great to me.  Nothing else comes to mind as missing.  Happy to merge this once @mark-petersen has a look.

**Review by mark-petersen — APPROVED** [review]
Looks good. Thanks!

**Comment by xylar** [comment]
@vanroekel, this is ready for you to merge when you're comfortable doing so.

**Pull request merged**
